File: GeneratedEntityReaderClasses/Components/HitEffectComponent.cs
namespace GeneratedNoitaClasses;
public class HitEffectComponent : NoitaComponentBase,  iNoitaType<HitEffectComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUInt condition_effect { get; set; } = new();
	public NoitaUInt condition_status { get; set; } = new();
	public NoitaUInt effect_hit { get; set; } = new();
	public NoitaInt value { get; set; } = new();
	public NoitaString value_string { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		condition_effect.Read(s);
		condition_status.Read(s);
		effect_hit.Read(s);
		value.Read(s);
		value_string.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		condition_effect.Write(s);
		condition_status.Write(s);
		effect_hit.Write(s);
		value.Write(s);
		value_string.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/HitEffectComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class HitEffectComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    condition_effect:NoitaUInt = field(default_factory=NoitaUInt)
    condition_status:NoitaUInt = field(default_factory=NoitaUInt)
    effect_hit:NoitaUInt = field(default_factory=NoitaUInt)
    value:NoitaInt = field(default_factory=NoitaInt)
    value_string:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.condition_effect.Read(s)
        self.condition_status.Read(s)
        self.effect_hit.Read(s)
        self.value.Read(s)
        self.value_string.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.condition_effect.Write(s)
        self.condition_status.Write(s)
        self.effect_hit.Write(s)
        self.value.Write(s)
        self.value_string.Write(s)

